service/matching/task_writer.go

```diff
@@ -227,29 +222,15 @@ writerLoop:
 			reqs = w.getWriteBatch(reqs)
 			batchSize := len(reqs)
 
-			maxReadLevel := int64(0)
-
 			taskIDs, err := w.allocTaskIDs(ctx, batchSize)
 			if err != nil {
 				w.sendWriteResponse(reqs, nil, err)
 				continue writerLoop
 			}
 
-			var tasks []*persistencespb.AllocatedTaskInfo
-			for i, req := range reqs {
-				tasks = append(tasks, &persistencespb.AllocatedTaskInfo{
-					TaskId: taskIDs[i],
-					Data:   req.taskInfo,
-				})
-				maxReadLevel = taskIDs[i]
-			}
-
-			resp, err := w.appendTasks(ctx, tasks)
+			resp, err := w.appendTasks(ctx, taskIDs, reqs)
 			// Update the maxReadLevel after the writes are completed, but before we send the response,
 			// so that taskReader is guaranteed to see the new read level when SpoolTask wakes it up.
-			if maxReadLevel > 0 {
```

Comment:
is this check dropped by intention?

Reply:
seemed to have missed the fact that if we don't check it, as being done here, it might set it to 0 and could break things. Shall add the case where it updates inside of `db`.

My bad!